validphys2/src/validphys/config.py

```diff
@@ -364,14 +365,32 @@ def produce_fitpdfandbasis(self, fitpdf, basisfromfit):
 
     @element_of("dataset_inputs")
     def parse_dataset_input(self, dataset: Mapping):
-        """The mapping that corresponds to the dataset specifications in the
-        fit files"""
-        known_keys = {"dataset", "sys", "cfac", "frac", "weight", "custom_group"}
+        """The mapping that corresponds to the dataset specifications in the fit files
+
+        This mapping is such that
+            dataset: str
+                name of the dataset to load
+            variant: str
+                variant of the dataset to load
+            cfac: list
+                list of cfactors to apply
+            frac: float
+                fraction of the data to consider for training purposes
+            weight: float
+                extra weight to give to the dataset
+            custom_group: str
+                custom group to apply to the dataset
+
+        Note that the `sys` key is deprecated and allowed only for old-format dataset.
+
+        Old-format commondata will be translated to the new version in this function.
+        """
+        accepted_keys = {"dataset", "sys", "cfac", "frac", "weight", "custom_group", "variant"}
         try:
             name = dataset["dataset"]
             if not isinstance(name, str):
                 raise ConfigError(f"'dataset' must be a string, not {type(name)}")
-            # Check whether this is an integrability or positivity dataset (in the only way we know?)
+            #! Check whether this is an integrability or positivity dataset (in the only way we know?)
             if name.startswith(("INTEG", "POS")):
```

Comment:
I guess these should also change.
```suggestion
            #! Check whether this is an integrability or positivity dataset (in the only way we know?)
            if name.startswith(("NNPDF_INTEG", "NNPDF_POS")):
```